Connect to AWS Athena using Sqlalchemy and PyAthena

fivetran-rishabhghosh opened the pull request.

varundhall commented on examples/source_examples/aws_athena/using_boto3/connector.py:
```diff
@@ -0,0 +1,125 @@
+    while True:
+        response = athena_client.get_query_results(QueryExecutionId=query_execution_id,
+                                                   NextToken=response['NextToken']) \
+            if 'NextToken' in response else athena_client.get_query_results(QueryExecutionId=query_execution_id)
```
nit: define as constant

varundhall commented on examples/source_examples/aws_athena/using_boto3/connector.py:
```diff
@@ -0,0 +1,125 @@
+                elif 'BooleanValue' in cell:
+                    row_data.append(bool(cell['BooleanValue']))
+                else:
+                    row_data.append(None)  # Handle unexpected data types
```
it should throw some exception?

fivetran-rishabhghosh replied:
Removed the comment. It handles null values.

varundhall commented on examples/source_examples/aws_athena/using_boto3/connector.py:
```diff
@@ -0,0 +1,125 @@
+                        region_name=configuration['region_name'])
+
+
+def get_query_results(athena_client, query_execution_id):
```
split this into small methods for better readability

varundhall commented on examples/source_examples/aws_athena/using_sqlAlchemy/connector.py:
```diff
@@ -0,0 +1,79 @@
+# This is a simple example for how to work with the fivetran_connector_sdk module.
```
please be consistent with casing in directory name `using_sql_alchemy`

fivetran-rishabhghosh replied:
should be SQLAlchemy

5tran-alexil commented on README.md:
```diff
@@ -114,6 +114,12 @@ Refer to the Multithreading Guidelines in api_threading_utils.py
+This is an example of how we can sync data from AWS Athena via Connector SDK.
+We have two examples, one utilises boto3 and another uses SQLAlchemy with PyAthena. 
+You can use either based on your requirements.
+You would need to provide your AWS Athena credentials for this example to work.
```
```suggestion
You need to provide your AWS Athena credentials for this example to work.
```

5tran-alexil commented on examples/source_examples/aws_athena/using_boto3/connector.py:
```diff
@@ -0,0 +1,128 @@
+# This is a simple example for how to work with the fivetran_connector_sdk module.
+# This is an example to show how we can sync records from AWS Athena via Connector SDK using boto3.
+# You would need to provide your credentials for this example to work.
```
```suggestion
# You need to provide your credentials for this example to work.
```

5tran-alexil commented on examples/source_examples/aws_athena/using_boto3/connector.py:
```diff
@@ -0,0 +1,128 @@
+NEXT_TOKEN = "NextToken"
+
+
+# Define the schema function which lets you configure the schema your connector delivers.
```
```suggestion
# Define the schema function, which lets you configure the schema your connector delivers.
```

5tran-alexil commented on examples/source_examples/aws_athena/using_boto3/connector.py:
```diff
@@ -0,0 +1,128 @@
+# See the technical reference documentation for more details on the update function
+# https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update
+# The function takes two parameters:
+# - configuration: dictionary contains any secrets or payloads you configure when deploying the connector
```
```suggestion
# - configuration: a dictionary that contains any secrets or payloads you configure when deploying the connector
```

5tran-alexil commented on examples/source_examples/aws_athena/using_boto3/connector.py:
```diff
@@ -0,0 +1,128 @@
+# https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update
+# The function takes two parameters:
+# - configuration: dictionary contains any secrets or payloads you configure when deploying the connector
+# - state: a dictionary contains whatever state you have chosen to checkpoint during the prior sync
```
```suggestion
# - state: a dictionary that contains whatever state you have chosen to checkpoint during the prior sync.
```

5tran-alexil commented on examples/source_examples/aws_athena/using_boto3/connector.py:
```diff
@@ -0,0 +1,128 @@
+# The function takes two parameters:
+# - configuration: dictionary contains any secrets or payloads you configure when deploying the connector
+# - state: a dictionary contains whatever state you have chosen to checkpoint during the prior sync
+# The state dictionary is empty for the first sync or for any full re-sync
```
```suggestion
# The state dictionary is empty for the first sync or for any full re-sync.
```

5tran-alexil commented on examples/source_examples/aws_athena/using_boto3/connector.py:
```diff
@@ -0,0 +1,128 @@
+        print(f"Query failed with status: {status}")
+
+
+# This creates the connector object that will use the update function defined in this connector.py file.
```
```suggestion
# This creates the connector object - connection - that will use the update function defined in this connector.py file.
```

fivetran-rishabhghosh replied:
This is incorrect. When we say a connector object we mean the Connector Object in python code referred in the sdk package

5tran-alexil commented on examples/source_examples/aws_athena/using_boto3/connector.py:
```diff
@@ -0,0 +1,128 @@
+# Check if the script is being run as the main module.
+# This is Python's standard entry method allowing your script to be run directly from the command line or IDE 'run' button.
+# This is useful for debugging while you write your code. Note this method is not called by Fivetran when executing your connector in production.
+# Please test using the Fivetran debug command prior to finalizing and deploying your connector.
```
```suggestion
# Test it using the Fivetran debug command prior to finalizing and deploying your connector.
```

5tran-alexil commented on examples/source_examples/aws_athena/using_sqlalchemy/connector.py:
```diff
@@ -0,0 +1,79 @@
+# This is a simple example for how to work with the fivetran_connector_sdk module.
```
```suggestion
# This is a simple example illustrating how to work with the fivetran_connector_sdk module.
```

5tran-alexil commented on examples/source_examples/aws_athena/using_sqlalchemy/connector.py:
```diff
@@ -0,0 +1,79 @@
+# This is a simple example for how to work with the fivetran_connector_sdk module.
+# This is an example to show how we can sync records from AWS Athena via Connector SDK using SQLAlchemy and PyAthena.
```
```suggestion
# This is an example to show how we can sync records from AWS Athena by using Connector SDK and SQLAlchemy and PyAthena.
```

5tran-alexil commented on examples/source_examples/aws_athena/using_sqlalchemy/connector.py:
```diff
@@ -0,0 +1,79 @@
+
+TABLE_NAME = "test_rows"
+
+# Define the schema function which lets you configure the schema your connector delivers.
```
```suggestion
# Define the schema function, which lets you configure the schema your connector delivers.
```

5tran-alexil commented on examples/source_examples/aws_athena/using_sqlalchemy/connector.py:
```diff
@@ -0,0 +1,79 @@
+# See the technical reference documentation for more details on the update function
+# https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update
+# The function takes two parameters:
+# - configuration: dictionary contains any secrets or payloads you configure when deploying the connector
```
```suggestion
# - configuration: a dictionary that contains any secrets or payloads you configure when deploying the connector
```

5tran-alexil commented on examples/source_examples/aws_athena/using_sqlalchemy/connector.py:
```diff
@@ -0,0 +1,79 @@
+# https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update
+# The function takes two parameters:
+# - configuration: dictionary contains any secrets or payloads you configure when deploying the connector
+# - state: a dictionary contains whatever state you have chosen to checkpoint during the prior sync
```
```suggestion
# - state: a dictionary that contains whatever state you have chosen to checkpoint during the prior sync
```

5tran-alexil commented on examples/source_examples/aws_athena/using_sqlalchemy/connector.py:
```diff
@@ -0,0 +1,79 @@
+# The function takes two parameters:
+# - configuration: dictionary contains any secrets or payloads you configure when deploying the connector
+# - state: a dictionary contains whatever state you have chosen to checkpoint during the prior sync
+# The state dictionary is empty for the first sync or for any full re-sync
```
```suggestion
# The state dictionary is empty for the first sync or for any full re-sync.
```

5tran-alexil commented on examples/source_examples/aws_athena/using_sqlalchemy/connector.py:
```diff
@@ -0,0 +1,79 @@
+                                })
+
+
+# This creates the connector object that will use the update function defined in this connector.py file.
```
```suggestion
# This creates the connector object - a connection - that will use the update function defined in this connector.py file.
```

fivetran-rishabhghosh replied:
This is incorrect. When we say a connector object we mean the Connector Object in python code referred in the sdk package

5tran-alexil commented on examples/source_examples/aws_athena/using_sqlalchemy/connector.py:
```diff
@@ -0,0 +1,79 @@
+# Check if the script is being run as the main module.
+# This is Python's standard entry method allowing your script to be run directly from the command line or IDE 'run' button.
+# This is useful for debugging while you write your code. Note this method is not called by Fivetran when executing your connector in production.
+# Please test using the Fivetran debug command prior to finalizing and deploying your connector.
```
```suggestion
# Test it by using the Fivetran debug command prior to finalizing and deploying your connector.
```

5tran-alexil reviewed (commented):
See my comments/suggestions.

fivetran-rishabhghosh commented:
@5tran-alexil I reverted the casing of the headings.
It should be consistent with the folder names in the repository.

Also, the ordering was messed up. Fixed that as well

varundhall commented on README.md:
```diff
@@ -16,23 +16,22 @@ There are several examples available under `/examples`:
   </summary>
```
please get the readme approved by Tech Writers

5tran-alexil reviewed (approved):
LGTM
